### PR TITLE
fix broken test and update configure to 2.71

### PR DESCRIPTION
**Pull request by jedwards4b**
Fixes issue #11 

The issue was that the recv datatype sent to the irecv call was ignored by the copy data called in the send call.  In order to fix this I added datatype and count to the Req structure so that they can be used in the send.

(@rljacob  added later) Also update the configure from 2.70 to 2.71.

### CODE REVIEW

File: recv.c [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
 static int mpi_match_send(void *r, void *tag)
 {
-  return( *((int *)tag) == MPI_ANY_TAG ||
+//    printf("mpi_match_send %d %d\n",*(int *)tag, ((Req *) r)->tag);
```
Comment (gold2718) [inline recv.c]: This can (and should IMHO) be removed.

File: send.c [hunk 2]
```diff
@@ -11,7 +11,8 @@
 
 static int mpi_match_recv(void *r, void *tag)
 {
-  return( ((Req *)r)->tag == MPI_ANY_TAG ||
+//    printf("mpi_match_recv %d %d\n",*((int *) tag), ((Req *)r)->tag);
```
Comment (gold2718) [inline send.c]: This can (and should IMHO) be removed.

File: send.c [hunk 3]
```diff
@@ -63,14 +64,14 @@ int MPI_Isend(void *buf, int count, MPI_Datatype datatype,
...
     }
-
   if (( match=AP_list_search_func(mycomm->recvlist,mpi_match_recv,&tag)))
     {
       rreq=(Req *)AP_listitem_data(match);
       AP_list_delete_item(mycomm->recvlist,match);
 
 //      memcpy(rreq->buf,buf,count * datatype);
-      copy_data2(buf, count, datatype, rreq->buf, count, datatype);
+//      printf("calling copy datatype %d %d %d %d\n",(int) datatype, count, rreq->type, rreq->count);
```
Comment (gold2718) [inline send.c]: Please remove all debugging print statements.

File: type.c [hunk 4]
```diff
@@ -220,6 +220,18 @@ int MPI_Type_extent(MPI_Datatype type, MPI_Aint * extent)
...
+/*
+int MPI_Type_get_envelope(MPI_Datatype type, int *num_integers, int *num_addresses, int *num_datatypes, int *combiner)
+{
+    *num_integers = 
+
+}
+
+*/
+
+
```
Comment (gold2718) [inline type.c]: Is this needed for anything? If not, please remove.

File: type.c [hunk 5]
```diff
@@ -821,7 +833,251 @@ int MPI_Type_free(MPI_Datatype * datatype)
...
+void printMapDatatype(MPI_Datatype datatype) {
+    MPI_Aint lb, ub;
+    MPI_Type_lb(datatype, &lb);
+    MPI_Type_ub(datatype, &ub);
+
+    printf("\"(LB, %ld), ", lb);
+    printdatatype(datatype, 0);
+    printf("(UB, %ld)\"\n", ub);
+}
+*/
```
Comment (gold2718) [inline type.c]: All this commented out code is confusing. Either remove it or maybe bracket it with something like `#ifdef DEBUG_PRINT`.

**Review by gold2718 — APPROVED** [review]
Thanks for the fix and the new tests!
I'm approving but it would be really nice to clean up the code a bit before merging.

**Comment by rljacob** [comment]
Why did this PR also change the configure?

**Pull request merged**